docs/guides/python/serverless-ai-api.mdx

```diff
@@ -0,0 +1,348 @@
+├── python.dockerfile
+├── python.dockerfile.dockerignore
+└── README.md
+```
+
+The `services/api.py` is a sample service that Nitric created (a simple “Hello, World” HTTP endpoint). We'll replace this with our own code. The `nitric.yaml` config by default tells Nitric to treat any `*.py` in `services/` as a service and what runtime to use. We'll keep the config mostly default.
+
+4. **(Optional) Verify the template works:** You can test the starter project right away by running `nitric start` inside the project. This spins up Nitric's local development server. Check the terminal output - you should see that an API endpoint (perhaps `/hello/:name`) is available locally at some port (like 4001). For now, stop the server (Ctrl+C) - we're going to write our own service next.
+
+With the project set up, let's start coding our API.
+
+## Writing the Python API Service (with AI Integration)
+
+Our goal is to create an HTTP endpoint `/summarize` (for example) that accepts some text and returns a summary of that text. We'll implement this as an **API service** in Nitric. In Nitric's model, an **API** can have multiple RESTful routes, served by one or more services. We define an API and then use decorators to attach function handlers to specific HTTP methods and paths - very similar to frameworks like Flask or FastAPI, but with Nitric's `@api` decorators.
+
+**Steps:**
+
+1. **Create a new service file:** In the `services/` directory, create a file `summarize.py` (you can name it differently, but let's go with this for clarity).
+
+2. **Import Nitric and dependencies:** We'll need Nitric's API utilities and the OpenAI library (for AI). Make sure you add `openai` to your dependencies (`uv add openai`). Then, in `summarize.py`:
+
+```python
+import os
+from openai import OpenAI
+from nitric.resources import api
+from nitric.application import Nitric
+from nitric.context import HttpContext
+```
+
+Here we import `api` to create an API resource, `Nitric` to run the app, and `HttpContext` for typing (the context object passed into handlers). We'll use `os` to read the OpenAI API key from an environment variable, and `openai` to call the OpenAI service.
+
+3. **Configure OpenAI API key:** It's best not to hardcode secrets. If you have your OpenAI API key, set it as an environment variable (e.g., create a .env file for the project containing `OPENAI_API_KEY=sk-...`). We can then configure the OpenAI client in code:
+
+```python
+# Configure OpenAI API key from env
+api_key = os.environ.get("OPENAI_API_KEY")
+if not api_key:
+  raise Exception("Please set the OPENAI_API_KEY environment variable")
+```
+
+This will fetch the API key from your environment. (If you prefer, Nitric also has a built-in **Secrets** feature to securely manage secrets, but for simplicity we'll stick to an env var here.)
+
+4. **Create the OpenAI client:** Now, use Nitric to create a POST route for summarization:
+
+```python
+# Create OpenAI client
+client = OpenAI(api_key=api_key)
+```
```

Comment:
Could also provide the user the option of testing locally using ollama if they have it installed